Sources/UIKitNavigation/Navigation/NavigationStackController.swift

```diff
@@ -24,7 +24,19 @@
 
     public override weak var delegate: (any UINavigationControllerDelegate)? {
       get { pathDelegate.base }
-      set { pathDelegate.base = newValue }
+      set {
+        if super.delegate != nil {
+          reportIssue(
+            """
+            UINavigationControllerDelegate object is being set after
+            NavigationStackController's viewDidLoad() has been called,
+            so delegate functions will not be observed. Set your delegate
+            in your subclass's initializer to fix this issue
+            """
+          )
```

Comment:
Isn't this an issue that plagues UIKit in general? Do we really need to report this issue in our library? People should just know that they always need to set their delegates right when the controller is initialized, and never do it later.